File: Python/2019/d2p1.py
import math

input  = open("input.txt", "r")

input = input.read()

inputList = input.split(',')

for i in range(len(inputList)):
    inputList[i] = int(inputList[i])

noun = 12

verb = 2

def run(arr, noun, verb):
    arr[1] = noun
    arr[2] = verb
    i = 0
    while(i < len(arr)):
        if(arr[i] == 1):
            arr[arr[i+3]] = arr[arr[i+1]] + arr[arr[i+2]]
        elif(arr[i] == 2):
            arr[arr[i+3]] = arr[arr[i+1]] * arr[arr[i+2]]
        elif(arr[i] == 99):
            return arr[0]
        i = i + 4

print(run(inputList, noun, verb))

File: CPP/2019/d2p1.cpp
#include <fstream>
#include <iostream>
#include <string>
#include <stdio.h>
#include <algorithm>
#include <vector>
#include <sstream>

using namespace std;

int main () {
    string data;

    // open a file in read mode.
    ifstream infile; 
    infile.open("input.txt"); 
    // puts file data into string called data
    infile >> data; 

    std::replace(data.begin(), data.end(), ',', ' ');

    vector<int> array;

    stringstream ss(data);

    int temp;

    while (ss >> temp)
        array.push_back(temp);

    for(int i=0; i < array.size(); i++)
        std::cout << array.at(i) << ' ';

}